addons/microsoft_outlook/views/ir_mail_server_views.xml

```diff
@@ -37,6 +42,9 @@
                     </div>
                 </div>
             </field>
+            <field name="smtp_pass" position="attributes">
+                <attribute name="attrs">{'invisible' : [('use_microsoft_outlook_service','=', True)]}</attribute>
+            </field>
```

Comment:
@thiryr 
Not sure if there is a trick to make the domain be `{'invisible' : ['|',('use_google_gmail_service',True),('use_microsoft_outlook_service','=', True)]}` depending on the fact if only one module or both modules are installed.

In this state, depending on which extended view comes first, the password will only be invisible for when the `use_google_gmail_service` or the `use_microsoft_outlook_service` checkbox is ticked.